src/proxy/http/profiles.rs

```diff
@@ -413,24 +525,37 @@ pub mod router {
         }
 
         fn call(&mut self, target: T) -> Self::Future {
-            let inner = self.inner.make(&target);
-            let stack = self.route_layer.clone().service(svc::shared(inner));
+            // Initially there are no dst_overrides, so build a concrete router
+            // with only the default target.
+            let mut make = IndexMap::with_capacity(1);
+            make.insert(target.clone(), self.inner.make(&target));
+
+            let concrete_router =
+                rt::Router::new_fixed(ConcreteDstRecognize::new(target.clone(), Vec::new()), make);
```

Comment:
tioli:  i prefer breaking up complex constructors into multiple lines so they're easier to modify later:

```rust
let concrete_router = {
  let rec = ConcreteDstRecognize(...)
  rt::Router::new_fixed(rec, make)
};
```